pkg/image/registry/imagestreammapping/rest.go

```diff
@@ -19,63 +19,85 @@ import (
 // result in resource conflicts.
 const maxRetriesOnConflict = 10
 
-// REST implements the RESTStorage interface in terms of an image registry and
-// image stream registry. It only supports the Create method and is used
-// to simplify adding a new Image and tag to an ImageStream.
-type REST struct {
-	imageRegistry       image.Registry
-	imageStreamRegistry imagestream.Registry
-}
-
-// NewREST returns a new REST.
-func NewREST(imageRegistry image.Registry, imageStreamRegistry imagestream.Registry) *REST {
-	return &REST{
-		imageRegistry:       imageRegistry,
-		imageStreamRegistry: imageStreamRegistry,
-	}
-}
-
-// imageStreamMappingStrategy implements behavior for image stream mappings.
-type imageStreamMappingStrategy struct {
+// Strategy implements behavior for image stream mappings.
+type Strategy struct {
```

Comment:
Can we use this refactor to move this `Strategy` into its own file? - the same way the other resources do.